File: ios/coffee-oc/coffee/coffee/DataModels/Bases.m
//
//  Bases.m
//  coffee
//
//  Created by achiel on 14-10-6.
//  Copyright (c) 2014年 1clip. All rights reserved.
//

#import "Bases.h"

@implementation IdObject

+(instancetype) idWithValue:(long) id
{
    IdObject* instance = [[IdObject alloc] init];
    instance.id = id;
    
    return instance;
}

+(JSONKeyMapper*) keyMapper
{
    return [JSONKeyMapper mapperFromUnderscoreCaseToCamelCase];
}

@end

@implementation AvatarInfo

@end

@implementation Character

@end

File: ios/coffee/coffee/Data Models/Bases.swift
//
//  Bases.swift
//  coffee
//
//  Created by achiel on 14-10-3.
//  Copyright (c) 2014年 1Clip. All rights reserved.
//

public class IdObject : JSONModel {
    public var id : NSString!
    
    public override class func keyMapper() -> JSONKeyMapper {
        return JSONKeyMapper.mapperFromUnderscoreCaseToCamelCase()
    }
    
}

public class AvatarInfo : IdObject {
    public var token : NSString!
    
    public override class func keyMapper() -> JSONKeyMapper {
        return JSONKeyMapper.mapperFromUnderscoreCaseToCamelCase()
    }
}

public class Character : IdObject {

    public var friendlyName: NSString!
    
    public var avatarInfo: AvatarInfo!
}